### ISSUE TITLE
Typescript mixin support [Bug Report]

### ISSUE THREAD

**Issue by yujiaze** [babel/babel#9839]
## Bug Report

**Current Behavior**

### Mixin method declaration becomes `undefined`

**Input Code**

```js
// Activatable Mixin
class Activatable {
    isActive: boolean;
    activate() {
        this.isActive = true;
    }
    deactivate() {
        this.isActive = false;
    }
}

class SmartObject implements Activatable {
    constructor() {
       
    }

    interact() {
        this.activate();
    }

    // Activatable
    isActive: boolean = false;
    activate: () => void;
    deactivate: () => void;
}
applyMixins(SmartObject, [Activatable]);

let smartObj = new SmartObject();
setTimeout(() => smartObj.interact(), 1000);

// Error: activate is not a method

////////////////////////////////////////
// In your runtime library somewhere
////////////////////////////////////////

function applyMixins(derivedCtor: any, baseCtors: any[]) {
    baseCtors.forEach(baseCtor => {
        Object.getOwnPropertyNames(baseCtor.prototype).forEach(name => {
            Object.defineProperty(derivedCtor.prototype, name, Object.getOwnPropertyDescriptor(baseCtor.prototype, name));
        });
    });
}

```

**Expected behavior/code**
A clear and concise description of what you expected to happen (or code).

**Babel Configuration (.babelrc)**

```js
{
    "presets": [
        "@babel/preset-typescript",
        [
            "@babel/preset-env",
            {
                "modules": false,
                "targets": {
                    "browsers": [
                        "> 1%",
                        "last 2 versions",
                        "ie >= 10"
                    ]
                }
            }
        ],
        "@babel/preset-react",
    ],
    "plugins": [
        "react-hot-loader/babel",
        "react-require",
        "@babel/plugin-proposal-object-rest-spread",
        [
            "@babel/plugin-proposal-decorators",
            {
                "legacy": true
            }
        ],
        [
            "@babel/plugin-proposal-class-properties",
            {
                "loose": true
            }
        ],
        "react-html-attrs",
        "jsx-control-statements",
        [
            "@babel/plugin-transform-runtime",
            {
                "corejs": 2
            }
        ],
        "@babel/plugin-syntax-dynamic-import",
        "@babel/plugin-syntax-import-meta",
        "@babel/plugin-proposal-json-strings",
        "@babel/plugin-proposal-function-sent",
        "@babel/plugin-proposal-export-namespace-from",
        "@babel/plugin-proposal-numeric-separator",
        "@babel/plugin-proposal-throw-expressions",
        "@babel/plugin-proposal-export-default-from",
        "@babel/plugin-proposal-logical-assignment-operators",
        "@babel/plugin-proposal-optional-chaining",
        [
            "@babel/plugin-proposal-pipeline-operator",
            {
                "proposal": "minimal"
            }
        ],
        "@babel/plugin-proposal-nullish-coalescing-operator",
        "@babel/plugin-proposal-do-expressions",
        "@babel/plugin-proposal-function-bind"
    ]
}
```

**Environment**
- Babel version(s):
```
        "@babel/core": "^7.0.0",
        "@babel/plugin-proposal-class-properties": "^7.0.0",
        "@babel/plugin-proposal-decorators": "^7.0.0",
        "@babel/plugin-proposal-do-expressions": "^7.0.0",
        "@babel/plugin-proposal-export-default-from": "^7.0.0",
        "@babel/plugin-proposal-export-namespace-from": "^7.0.0",
        "@babel/plugin-proposal-function-bind": "^7.0.0",
        "@babel/plugin-proposal-function-sent": "^7.0.0",
        "@babel/plugin-proposal-json-strings": "^7.0.0",
        "@babel/plugin-proposal-logical-assignment-operators": "^7.0.0",
        "@babel/plugin-proposal-nullish-coalescing-operator": "^7.0.0",
        "@babel/plugin-proposal-numeric-separator": "^7.0.0",
        "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
        "@babel/plugin-proposal-optional-chaining": "^7.0.0",
        "@babel/plugin-proposal-pipeline-operator": "^7.0.0",
        "@babel/plugin-proposal-throw-expressions": "^7.0.0",
        "@babel/plugin-syntax-dynamic-import": "^7.0.0",
        "@babel/plugin-syntax-import-meta": "^7.0.0",
        "@babel/plugin-transform-runtime": "^7.0.0",
        "@babel/preset-env": "^7.0.0",
        "@babel/preset-react": "^7.0.0",
        "@babel/preset-typescript": "^7.3.3",
```
- Node/npm version: [Node 10.14.1/npm 6.4.1]
- OS: OSX 10.12.5
- Monorepo: no
- How you are using Babel: `loader`

**Possible Solution**

remove all the declaration, not only the types

**Additional context/Screenshots**

compile typescript using babel-loader :

SmartObject
  - activate: undefind
  - deactivate: undefind
  - __proto__
     - activate()
     - deactivate()

compile typescript using ts-loader :

SmartObject
  - __proto__
     - activate()
     - deactivate()

**Comment by nicolo-ribaudo** [comment]
Duplicate of https://github.com/babel/babel/issues/9105

TL;DR Babel follows the class properties proposal, while typescript doesn't: https://github.com/Microsoft/TypeScript/issues/12437
